File: src/memory.py
from typing import Dict, List

class Memory:
    def __init__(self):
        self.short_term = []
        self.long_term = {}
    
    def add(self, item: Dict):
        self.short_term.append(item)
        
    def get_recent_context(self) -> List[Dict]:
        return self.short_term[-5:]  # Last 5 interactions 

File: src/lib/memory.ts
import { Message } from '../types/agent';

class Memory {
  private shortTerm: Message[];
  private longTerm: Map<string, any>;

  constructor() {
    this.shortTerm = [];
    this.longTerm = new Map();
  }

  add(message: Message): void {
    this.shortTerm.push(message);
  }

  getRecentContext(limit: number = 5): Message[] {
    return this.shortTerm.slice(-limit);
  }
}

export default Memory; 